chain/ethereum/src/data_source.rs

```diff
@@ -602,15 +602,25 @@ impl DataSource {
                 // Take the input for the call, chop off the first 4 bytes, then call
                 // `function.decode_input` to get a vector of `Token`s. Match the `Token`s
                 // with the `Param`s in `function.inputs` to create a `Vec<LogParam>`.
-                let tokens = function_abi
+                let tokens = match function_abi
                     .decode_input(&call.input.0[4..])
                     .with_context(|| {
                         format!(
                             "Generating function inputs for the call {:?} failed, raw input: {}",
                             &function_abi,
                             hex::encode(&call.input.0)
                         )
-                    })?;
+                    }) {
+                      Ok(val) => val,
+                      Err(err) => {
+                        warn!(logger, "Failed parsing inputs, skipping"; "error" => &err.to_string());
+                        Vec::new()
+                      },
+                    };
+                
+                if tokens.len() == 0 {
```

Comment:
I'm just not sure if we should do this validation here or in the `adapter`, like I've done [here](https://github.com/graphprotocol/graph-node/pull/3219/files#diff-2bd955f61f661e1160bebb53dd34de5cfc4200f5900c169cbbe7a0f3471c932eR333).

What do you think @leoyvens ?

Reply:
Thanks, @otaviopace for the review. The only reason I had to add the validation here is that the next line `ensure(tokens.len() == function_abi.inputs.len()` fails due to the length of tokens being 0. So instead of returning a None to continue, it was failing completely.

Reply:
It looks like this would `None` if the function has no arguments which is incorrect.

Reply:
Thanks. Removed it. :)